src/tools/crawler/main.rs

```diff
@@ -66,6 +82,20 @@ async fn main() {
     let crawler = Crawler::new().await;
 
     let mut network_metrics = NetworkMetrics::default();
+    let summary_snapshot = Arc::new(Mutex::new(NetworkSummary::default()));
+
+    if args.rpc {
+        // Initialize the RPC server if address is specified.
+        if let Some(addr) = args.bind_rpc {
+            let rpc_addr: SocketAddr = addr.parse().unwrap();
+            let rpc_context = RpcContext::new(Arc::clone(&summary_snapshot));
+            tokio::spawn(async move {
+                initialize_rpc_server(rpc_addr, rpc_context).await;
+            });
```

Comment:
any RPC server task should be spawned within `initialize_rpc_server` (and not here)